packages/dids/src/methods/did-dht.ts

```diff
@@ -0,0 +1,1568 @@
+    didUri: string
+  }): Promise<Jwk> {
+    // Parse the DID URI.
+    const parsedDid = DidUri.parse(didUri);
+
+    // Verify that the DID URI is valid.
+    if (!parsedDid) {
+      throw new DidError(DidErrorCode.InvalidDid, `Invalid DID URI: ${didUri}`);
+    }
+
+    // Verify the DID method is supported.
+    if (parsedDid.method !== DidDht.methodName) {
+      throw new DidError(DidErrorCode.MethodNotSupported, `Method not supported: ${parsedDid.method}`);
+    }
+
+    // Decode the method-specific identifier from z-base-32 to a byte array.
+    let identityKeyBytes: Uint8Array | undefined;
+    try {
+      identityKeyBytes = Convert.base32Z(parsedDid.id).toUint8Array();
+    } catch { /* Capture error */ }
+
+    // Verify that the method-specific identifier was decoded successfully.
+    if (!identityKeyBytes || identityKeyBytes.length !== 32) {
+      throw new DidError(DidErrorCode.InvalidDid, `Failed to decode method-specific identifier`);
+    }
+
+    // Convert the byte array to a JWK.
+    const identityKey = await Ed25519.bytesToPublicKey({ publicKeyBytes: identityKeyBytes });
+
+    return identityKey;
+  }
+
+  /**
+   * Encodes a DID DHT Identity Key into a DID identifier.
+   *
+   * This method first z-base-32 encodes the Identity Key. The resulting string is prefixed with
+   * `did:dht:` to form the DID identifier.
+   *
+   * @param params - The parameters to use for the conversion.
+   * @param params.identityKey The Identity Key from which the DID identifier is computed.
+   * @returns A promise that resolves to a string containing the DID identifier.
+   */
+  public static async identityKeyToIdentifier({ identityKey }: {
+    identityKey: Jwk
+  }): Promise<string> {
+    // Convert the key from JWK format to a byte array.
+    const publicKeyBytes = await Ed25519.publicKeyToBytes({ publicKey: identityKey });
+
+    // Encode the byte array as a z-base-32 string.
+    const identifier = Convert.uint8Array(publicKeyBytes).toBase32Z();
+
+    return `did:${DidDht.methodName}:${identifier}`;
+  }
+
+  /**
+   * Converts a DID URI to the byte array representation of the Identity Key.
+   *
+   * @param params - The parameters to use for the conversion.
+   * @param params.didUri - The DID URI containing the Identity Key.
+   * @returns A byte array representation of the Identity Key.
+   */
+  public static identifierToIdentityKeyBytes({ didUri }: {
+    didUri: string
+  }): Uint8Array {
+    // Parse the DID URI.
+    const parsedDid = DidUri.parse(didUri);
+
+    // Verify that the DID URI is valid.
+    if (!(parsedDid && parsedDid.method === DidDht.methodName)) {
+      throw new DidError(DidErrorCode.InvalidDid, `Invalid DID URI: ${didUri}`);
+    }
+
+    // Decode the method-specific identifier from z-base-32 to a byte array.
+    let identityKeyBytes: Uint8Array | undefined;
+    try {
+      identityKeyBytes = Convert.base32Z(parsedDid.id).toUint8Array();
+    } catch { /* Capture error */ }
```

Comment:
Do you want to throw new DidError here?

Reply:
The error is thrown a few lines down.  This was partly there as a type guard in a slightly different prior implementation.

I thought of a way to refactor and reduce some duplication so pushing up a commit.

Thanks for the second pair of eyes! 😄